jasmine/jasmine-standalone-3.1.0/src/Score.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+function Score() {
+  // this array logs the score after each frame
+  this._totalScore = [];
+
+  // this array logs the score after each roll
+  this._rollScore = [];
+  this._bonus = [];
+}
+
+Score.prototype.storeRollScore = function(kockedDownPins) {
+  this._rollScore.push(kockedDownPins);
+};
+
+Score.prototype.storeScore = function(kockedDownPins) {
+   this._totalScore.push(kockedDownPins);
+};
+
+Score.prototype.getScore = function() {
+  var counter = 0,
+   sum = 0;
+
+  for (counter = this._totalScore.length; counter--;) {
+    sum += this._totalScore[counter];
+  }
+  return sum;
+};
+
+Score.prototype.getScoreCard = function() {
+   return this._rollScore;
+};
+
+Score.prototype.checkBonus = function(index) {
+
+  if (this.isSixthRollStrike(index)) {
+    if (this.isFourthRollStrike(index)) {
+      this.addBonus(index);
+    }
+  } else if (this.isFourthRollStrike(index)) {
+    this.addBonus(index);
+  } else if (this._rollScore[index - 4] +
+    this._rollScore[index - 3] === 10) {
+      this.addBonus(index);
+  }
+};
+
+Score.prototype.addBonus = function(index) {
+  var bonus = 0;
```

Comment:
This method feels unnecessarily complex. Wherever you have magic numbers in your code, you should be thinking 'can I simplify this?'. By magic numbers, I mean that you have different logic depending on whether a value is a 1,2,3,4 etc. 

Applying a strike bonus should be the same in every case - why have different methods depending on whether it is a second roll, fourth roll or sixth roll?